syntax = "proto3";

package chimera;

service AgentService {
  rpc ListAgents (ListAgentsRequest) returns (ListAgentsResponse) {}
  rpc ExecuteTask (ExecuteTaskRequest) returns (ExecuteTaskResponse) {}
}

message ListAgentsRequest {}

message ListAgentsResponse {
  repeated string agent_names = 1;
}

message ExecuteTaskRequest {
  string agent_name = 1;
  string task = 2;
}

message ExecuteTaskResponse {
  string response = 1;
}
